builder/builder-next/exporter/export.go

```diff
@@ -73,8 +72,21 @@ func (e *imageExporterInstance) Name() string {
 	return "exporting to image"
 }
 
-func (e *imageExporterInstance) Export(ctx context.Context, ref cache.ImmutableRef, opt map[string][]byte) (map[string]string, error) {
-	if config, ok := opt[exporterImageConfig]; ok {
+func (e *imageExporterInstance) Export(ctx context.Context, inp exporter.Source) (map[string]string, error) {
+
+	if len(inp.Refs) > 1 {
+		return nil, fmt.Errorf("exporting multiple references to image store is currently unsupported")
+	}
+	ref := inp.Ref
+	if ref != nil && len(inp.Refs) == 1 {
+		return nil, fmt.Errorf("invalid exporter input: Ref and Refs are mutually exclusive")
+	}
+	// only one loop
+	for _, v := range inp.Refs {
+		ref = v
```

Comment:
Same here; perhaps (clearer that we're using the last value)

```go
ref = inp.Refs[:len(inp.Refs)-1]
```


Or (which would be equivalent with only one item in there)

```go
ref = inp.Refs[:0]
```

Reply:
Oh ... actually; so you're using `for .. range` so that it will only be executed _at most_ once.. i.e., if `inp.Refs` was empty, we'd preserve the value of `inp.Ref`.

Perhaps something like;

```go
ref := inp.Ref
if len(inp.Refs) > 0 {
    ref = inp.Refs[:0]
}
```

Reply:
I'm for ranging because it's a map otherwise I would have done something like what you're suggesting.